src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -629,15 +629,24 @@ def __call__(
                     if callback is not None and i % callback_steps == 0:
                         callback(i, t, latents)
 
-        # 8. Post-processing
-        image = self.decode_latents(latents)
+        if output_type == "latent":
+            image = latents
+            has_nsfw_concept = None
+        elif output_type == "pil":
+            # 8. Post-processing
+            image = self.decode_latents(latents)
 
-        # 9. Run safety checker
-        image, has_nsfw_concept = self.run_safety_checker(image, device, prompt_embeds.dtype)
+            # 9. Run safety checker
+            image, has_nsfw_concept = self.run_safety_checker(image, device, prompt_embeds.dtype)
 
-        # 10. Convert to PIL
-        if output_type == "pil":
+            # 10. Convert to PIL
             image = self.numpy_to_pil(image)
+        else:
```

Comment:
We should maybe in the future force the user to write `output_type="numpy"` or `output_type="np"` . At the moment the user can just write whatever and then numpy is used cc @patil-suraj @pcuenca

Reply:
Yes, it makes sense!